### DISCUSSION TITLE
Puma::Cluster.worker taking 40s every request

### DISCUSSION THREAD

**Discussion by Chris24680** [puma/puma#3275] [Q&A]
I have a Rails 6.1.5.1 app that I'm deploying on a new local environment using Puma 3.12.2. For some reason on this environment, every page load takes much longer than it should compared to the same app running in other environments. I installed rack-mini-profiler to see what's going on and have noticed something strange going on with Puma. 

[Flamegraph_for_payroll.speedscope.json](https://github.com/puma/puma/files/13297766/Flamegraph_for_payroll.speedscope.json)

Looking at the provided flame graph you can see that 40s of the page load time is taken up by `Puma::Cluster.worker`. I have no idea why this would be. I also don't know if Puma is the cause of the issue or if the flamegraph just looks this way due to how mini profiler handles multiple threads. 

Any clues on what to look for would be greatly appreciated.

**Comment by dentarg** [comment]
> Puma 3.12.2

😳

Try the latest Puma and see if you still have issues

If you still have issues, you will need to share an reproducible example (config, code) for anyone to help out

**Comment by Chris24680** [comment]
I've solved the problem. Turned out there was a misconfiguration for Redis which meant it was trying to connect to a ip that didn't exist, so it would hang for 40s. So it wasn't a problem with puma despite the profiling making it look like it was.
